lib/client/ipam_block_reader_writer.go

```diff
@@ -56,31 +56,41 @@ func (rw blockReaderWriter) getAffineBlocks(host string, ver ipVersion, pool *cn
 	for _, o := range datastoreObjs {
 		k := o.Key.(model.BlockAffinityKey)
 
-		// Add the block if no IP pool was specified, or if an IP pool was specified
-		// and the block falls within the given IP pool.
-		if pool == nil || pool.Contains(k.CIDR.IPNet.IP) {
+		// Add the block if no IP pools were specified, or if IP pools were specified
+		// and the block falls within the given IP pools.
+		if len(pools) == 0 {
 			ids = append(ids, k.CIDR)
+			continue
+		}
```

Comment:
I think I'd like this better if we used an if/else here rather than the if / continue.